### PR TITLE
verify account deployment even when nonce is 0

### PR DESCRIPTION
**Pull request by talhamalik883**
An RPC call is made to verify whether the smart account is deployed when the nonce is zero. We previously implemented logic to include initCode in userOp only when the nonce is zero. However, we discovered a scenario where someone might have deployed the smart account on your behalf, resulting in a nonce of zero. In this case, the initCode should not be added to userOp.

### CODE REVIEW

**Comment by livingrockrises** [comment]
don't merge this to master. please update the base to development.

File: packages/account/src/BiconomySmartAccount.ts [hunk 1]
```diff
@@ -288,10 +288,14 @@ export class BiconomySmartAccount extends SmartAccount implements IBiconomySmart
     } catch (error) {
       // Not throwing this error as nonce would be 0 if this.nonce() throw exception, which is expected flow for undeployed account
     }
+    let isDeployed = true
+
+    isDeployed = nonce.eq(0) ? await this.isAccountDeployed(this.address) : true
```
Comment (livingrockrises) [inline packages/account/src/BiconomySmartAccount.ts:293]:
```
if(nonce.eq(0)) {
 isDeployed = await this.isAccountDeployed(this.address)
}
```

above code would be more readable. lmk what you think

**Review by livingrockrises — COMMENTED** [review]
reviewed

File: packages/account/src/BiconomySmartAccount.ts (hunk 1 above)
Comment (talhamalik883) [reply packages/account/src/BiconomySmartAccount.ts:293]: > ```
> if(nonce.eq(0)) {
>  isDeployed = await this.isAccountDeployed(this.address)
> }
> ```
> 
> above code would be more readable. lmk what you think

yeah, i think yours approach is more readable. I was thinking of multiple cases and used ternary but removed other cases as they were not valid and did not changed the logic from ternary approach to if statement

**Review by livingrockrises — APPROVED** [review]
LGTM

**Pull request merged**